packages/runtime-common/indexer.ts

```diff
@@ -339,6 +344,33 @@ export class Indexer {
     ]);
   }
 
+  private rangeCondition(filter: RangeFilter, on: CodeRef): CardExpression {
+    on = filter.on ?? on;
+    return every([
+      this.typeCondition(on),
+      ...flatten(
```

Comment:
Instead of using `flatten`, just splat `Object.entries`. Like we do in eqCondition and likeCondition above: `…Object.entries(  )`. It makes it easier to read the DSL that way